addons/mail/models/mail_scheduled_message.py

```diff
@@ -167,6 +167,21 @@ def post_message(self):
         else:
             raise UserError(_("You are not allowed to send this scheduled message"))
 
+    def _prepare_message_post_values(self, scheduled_message):
```

Comment:
Are you giving a scheduled message parameter to a scheduled message method :D ?
```suggestion
    def _prepare_message_post_values(self):
        self.ensure_one()
```
Also, if you don't want to call ``_notification_parameters_whitelist`` each time, just use it as post filter in ``_post_message`` : generate everything, then remove non whitelisted stuff. It is going to make prepare method a bit easier to handle.